stuf_zds_payments/plugin.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from typing import Any
+
+from django.template.loader import render_to_string
+from django.utils.translation import gettext_lazy as _
+
+from openforms.registrations.contrib.stuf_zds.plugin import (
+    StufZDSRegistration,
+    ZaakOptionsSerializer,
+)
+from openforms.registrations.registry import register
+from openforms.submissions.models import Submission
+from openforms.template import render_from_string
+from openforms.template.validators import DjangoTemplateValidator
+from openforms.variables.service import get_static_variables
+from rest_framework import serializers
+from stuf.stuf_zds.client import ZaakOptions
+
+from .client import get_client
+from .registration_variables import register as variables_registry
+
+logger = logging.getLogger(__name__)
+
+PLUGIN_IDENTIFIER = "stuf-zds-create-zaak:ext-utrecht"
+
+
+def get_payment_status_update_text() -> str:
+    return render_to_string(
+        "registrations/contrib/stuf_zds_payments/payment_status_update_xml.txt"
+    ).strip()
+
+
+def prepare_value(value: Any):
+    match value:
+        case bool():
+            return "true" if value else "false"
+        case list():
+            return " ".join(value)
+        case float():
+            return str(value)
+        case _:
+            return value
+
+
+class ZaakPaymentOptionsSerializer(ZaakOptionsSerializer):
+    payment_status_update_xml = serializers.CharField(
+        label=_("payment status update XML template"),
+        help_text=_(
+            "This template is evaluated with the submission data and the resulting XML "
+            "is sent to the StUF-ZDS with a PATCH to update the payment field."
```

Comment:
```suggestion
            "This template is evaluated with the submission data and the resulting XML "
            "is included in extraElementen."
```
the template will need to be upated, since I'm concerned that we'd be overwriting other extraElementen - any form fields that are not explicitly mapped to a registration attribute are included in the extra_data (via `get_unmapped_data`)